persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/util/CacheMapUtil.java

```diff
@@ -89,11 +89,17 @@
     }
 
     @PostConstruct
-    private void initializeCacheMemory() {
+    private void init() {
+        initializeCacheMemory();
+    }
+
+    public void initializeCacheMemory() {
+        
         // CHANGES TO THIS LIST MUST BE PROPAGATED TO 'GlobalProperties'
         this.cacheEnabled = (!authenticate.isEmpty() 
                 && !authenticate.equals("false") 
                 && !authenticate.contains("social_auth"));
+        
         if (cacheEnabled) {
```

Comment:
Line 169:
```java
if (mp.getMolecularAlterationType().toString() == EntityType.GENERIC_ASSAY.toString()) {
```
How is this working? Are we just getting lucky with the GC's string deduplication here? This should probably be changed to 
```java
if (EntityType.GENERIC_ASSAY.toString().equals(mp.getMolecularAlterationType().toString())) {
```

Reply:
I did not write this code, but this indeed seems strange. I guess we have to ask @dippindots ?

Reply:
@pvannierop @Luke-Sikina This is definitely not correct, I think we should go with @Luke-Sikina's suggestion.

Reply:
Ok, updated to code suggested by Luke.